lib/audumbla/enrichments/coarse_geocode.rb

```diff
@@ -0,0 +1,203 @@
+require 'twofishes'
+require 'geokit'
+
+module Audumbla::Enrichments
+  ##
+  # Enriches a `DPLA::MAP::Place` node by running its data through external 
+  # geocoders, using heuristics to determine a matching feature from GeoNames,
+  # and repopulating the `Place` with related data.
+  #
+  # If the existing `Place` contains data other than a `providedLabel`, that 
+  # data will be used as context for evaluating interpretations. For example: 
+  # a `Place` with an existing latitude and longitude will verify that the
+  # point is within the bounding box for a candidate match.
+  # 
+  # `skos:exactMatch` are reserved for the GeoNames features returned by the 
+  # geocoder. Other matching URIs (currently: LC authorities) are included as
+  # `skos:closeMatch`
+  #
+  # @example enriching from a `#providedLabel`
+  #   
+  #    place = DPLA::MAP::Place.new.tap { |p| p.providedLabel = 'Georgia' }
+  #    CoarseGeocode.new.enrich_value.dump :ttl
+  #    # [
+  #    #    a <http://www.europeana.eu/schemas/edm/Place>;
+  #    #    <http://dp.la/about/map/providedLabel> "Georgia";
+  #    #    <http://www.geonames.org/ontology#countryCode> "US";
+  #    #    <http://www.w3.org/2003/01/geo/wgs84_pos#lat> 3.275042e1;
+  #    #    <http://www.w3.org/2003/01/geo/wgs84_pos#long> -8.350018e1;
+  #    #    <http://www.w3.org/2004/02/skos/core#closeMatch> <http://id.loc.gov/authorities/names/n79023113>;
+  #    #    <http://www.w3.org/2004/02/skos/core#exactMatch> <http://sws.geonames.org/4197000/>;
+  #    #    <http://www.w3.org/2004/02/skos/core#prefLabel> "Georgia, United States"
+  #    # ] .
+  #
+  # @example enriching from a `#providedLabel` with lat/lng guidance
+  #   
+  #    place = DPLA::MAP::Place.new.tap do |p| 
+  #      p.providedLabel = 'Georgia'
+  #      p.lat = 41.9997
+  #      p.long = 43.4998
+  #    end
+  #    
+  #    CoarseGeocode.new.enrich_value.dump :ttl
+  #    # [
+  #    #    a <http://www.europeana.eu/schemas/edm/Place>;
+  #    #    <http://dp.la/about/map/providedLabel> "Georgia";
+  #    #    <http://www.geonames.org/ontology#countryCode> "GE";
+  #    #    <http://www.w3.org/2003/01/geo/wgs84_pos#lat> 4.199998e1;
```

Comment:
These are [xsd:double](http://www.w3.org/2001/XMLSchema#double)s; [Turtle squashes](http://www.w3.org/TeamSubmission/turtle/#abbrev) these down to exponential notation for compactness (saves specifying a literal type).

The representation in Ruby is a simple Float, e.g.:

```ruby
result.lat.first
# => 32.75042
```

`wgs84_pos:lat/lng` call for "decimal degrees".